pkg/internal/meshdetect/detector.go

```diff
@@ -293,40 +293,40 @@ func (d *Detector) listAllSevices(ctx context.Context, pageSize int) ([]*corev1.
 	return serviceList, nil
 }
 
-// listAllEndpoints lists all endpoints in all namespaces.
-// returns map: namespaced name of endpoints -> endpoints resource
+// listAllEndpointsPerService lists all Endpoints per every Service in every namespaces.
+// returns map: namespaced name of Service -> slice of Endpoint resources
 //
 // example: client.ObjectKey{Namespace: "default", Name: "service1"} ->
 //
-//	&corev1.Endpoints{
-//	   ObjectMeta: metav1.ObjectMeta {
-//	     Namespace: "default",
-//	     Name: "service1", ...
-//	   },
-//	   Subsets: ...,
-//	   ...
+//	[]discoveryv1.Endpoint{
+//		{ TargetRef: &v1.ObjectReference{Kind: "Pod", Namespace: "default", Name: "pod1"...}... },
+//	    { TargetRef: &v1.ObjectReference{Kind: "Pod", Namespace: "default", Name: "pod2"...}... },
+//	    ...
 //	}.
-func (d *Detector) listAllEndpoints(ctx context.Context, pageSize int) (
-	map[client.ObjectKey]*corev1.Endpoints, error,
+func (d *Detector) listAllEndpointsPerService(ctx context.Context, pageSize int) (
+	map[client.ObjectKey][]discoveryv1.Endpoint, error,
 ) {
-	endpointsMap := map[client.ObjectKey]*corev1.Endpoints{}
-	continueToken := ""
+	endpointsMap := map[client.ObjectKey][]discoveryv1.Endpoint{}
+	var continueToken string
 	for {
-		partialEndpointsList := &corev1.EndpointsList{}
-		err := d.Client.List(ctx, partialEndpointsList, client.Limit(pageSize), client.Continue(continueToken))
-		if err != nil {
+		partialEndpointSliceList := &discoveryv1.EndpointSliceList{}
```

Comment:
nit: any particular reason to name this like that? It would be more idiomatic to name it a short and succinct name like e.g. `endpoints`. Its enclosed in the `for` loop's scope anyway